packages/orn-locator/src/index.ts

```diff
@@ -24,3 +24,51 @@ export const locateAll = (orn: string[]) => {
 };
 
 export default {locate, locateAll};
+
+
+type Unpromise<T> = T extends Promise<infer I> ? I : never;
```

Comment:
There's a standard `Awaited` type that does this, but idk if it matches the signature exactly.